app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -1680,6 +1684,38 @@ class PluginApi {
   addTopicsSectionLink(arg) {
     addSectionLink(arg);
   }
+
+  /**
+   * Decorate inline bbcode in to-markdown function.
+   *
+   * ```
+   * api.beforeToMarkdownTagDecorate(function (text) {
+   *   if (this.element.attributes.class === "spoiled") {
+   *     this.prefix = "[spoiler]";
+   *     this.suffix = "[/spoiler]";
+   *   }
+   * });
+   * ```
+   */
```

Comment:
Just some general sentiments but after reading the description here, I can't really tell what purpose the API serves and what I can do with this API as an end user.